quadriga/src/main/java/edu/asu/diging/quadriga/api/v1/GetCollectionsApiController.java

```diff
@@ -0,0 +1,33 @@
+package edu.asu.diging.quadriga.api.v1;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+
+import edu.asu.diging.quadriga.config.web.TokenInfo;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+
+@Controller
+public class GetCollectionsApiController {
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    @GetMapping("/api/v1/collections")
+    ResponseEntity<List<Collection>> getCollections(
+            @RequestHeader(name = "Authorization", required = false) String authHeader, Authentication authentication) {
+        if (!(authentication.getDetails() instanceof TokenInfo)) {
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+        TokenInfo tokenInfo = (TokenInfo) authentication.getDetails();
```

Comment:
I would try to put this logic into an aspect since this will be required by more than just this one controller in the future. The aspect should inject the tokeninfo on successful authorization and otherwise return unauthorized for now.